### PR TITLE
Brings back ShallowWater regression test

### LINKED ISSUE #2922: Shallow water regression tests often fail on GPU
This isn't a huge issue but I've noticed recently that often the shallow water regression tests on GPU fail with the following error:

```
Shallow Water Bickley jet simulation [GPU, ConservativeFormulation]: Test Failed at /net/ocean/home/data44/data5/glwagner/.buildkite-agent/builds/sverdrup-6/clima/oceananigans/test/regression_tests/shallow_water_bickley_jet_regression.jl:93
  | Expression: all(test_fields.v .≈ truth_fields.v)
  | Stacktrace:
  | [1] macro expansion
  | @ /net/ocean/home/data44/data5/glwagner/julia-1.8.5/share/julia/stdlib/v1.8/Test/src/Test.jl:464 [inlined]
  | [2] run_shallow_water_regression(arch::GPU, formulation::ConservativeFormulation; regenerate_data::Bool)
  | @ Main /net/ocean/home/data44/data5/glwagner/.buildkite-agent/builds/sverdrup-6/clima/oceananigans/test/regression_tests/shallow_water_bickley_jet_regression.jl:93
```

Sometimes I need to restart the test twice for it to pass. Like I said, it's not a huge issue, but do we understand why that's happening? Is there anything we can do to prevent that behavior?

### PR DESCRIPTION
**Pull request by navidcy**
Closes #2922

### CODE REVIEW

File: test/regression_tests/shallow_water_bickley_jet_regression.jl [hunk 1]
```diff
@@ -40,7 +41,7 @@ function run_shallow_water_regression(arch, formulation; regenerate_data = false
     end
 
     stop_iteration = 20
-    Δt = 1e-2
+    Δt = 5e-3
```
Comment (glwagner) [inline test/regression_tests/shallow_water_bickley_jet_regression.jl:44]:
```suggestion
    Δt = 1e-6
```

Comment (navidcy) [reply test/regression_tests/shallow_water_bickley_jet_regression.jl:44]: if 5e-3 is not enough I'll reduce it to 1e-6

Comment (glwagner) [reply test/regression_tests/shallow_water_bickley_jet_regression.jl:44]: Word. No need to cut it close at all here

**Comment by navidcy** [comment]
@simone-silvestri, the [GPU Regression test passes](https://buildkite.com/clima/oceananigans/builds/10213#01867f84-d9b6-454a-8d3e-d7062858ec81)! 🎉🪅🪅

**Review by simone-silvestri — APPROVED** [review]

**Comment by simone-silvestri** [comment]
Nice!

**Pull request merged**